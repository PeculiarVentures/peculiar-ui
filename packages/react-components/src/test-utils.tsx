import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
