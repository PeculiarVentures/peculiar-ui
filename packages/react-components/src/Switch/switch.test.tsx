import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Switch } from './index';

describe('<Switch />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <Switch id="test-id" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <Switch className="my-class-name" id="test-id" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('colors', () => {
    const colors: React.ComponentProps<typeof Switch>['color'][] = [
      'primary',
      'secondary',
    ];

    colors.forEach((color) => {
      it(`color "${color}"`, () => {
        const { asFragment } = render(
          <Switch color={color} id="test-id" />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
