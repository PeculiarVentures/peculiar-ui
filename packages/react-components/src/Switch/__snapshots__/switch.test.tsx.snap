// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> colors color "primary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-7);
  height: var(--pv-size-base-4);
  border-radius: var(--pv-size-base-4);
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  overflow: hidden;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: var(--pv-color-gray-6);
}

.emotion-1:checked {
  background-color: var(--pv-color-primary);
}

.emotion-1:checked+[aria-hidden] {
  -webkit-transform: translateX(calc(50% - 2px));
  -moz-transform: translateX(calc(50% - 2px));
  -ms-transform: translateX(calc(50% - 2px));
  transform: translateX(calc(50% - 2px));
}

.emotion-1:checked+[aria-hidden]:before {
  background-color: var(--pv-color-primary-shade-2);
}

.emotion-1:disabled {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.4;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):hover+[aria-hidden]:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus+[aria-hidden]:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active+[aria-hidden]:before {
  opacity: 0.3;
}

.emotion-2 {
  background: var(--pv-color-primary-contrast);
  border-color: var(--pv-color-gray-3);
  display: block;
  position: absolute;
  width: var(--pv-size-base-3);
  height: var(--pv-size-base-3);
  -webkit-transition: -webkit-transform 200ms;
  transition: transform 200ms;
  box-shadow: var(--pv-shadow-light-low);
  -webkit-transform: translateX(calc(-50% + 2px));
  -moz-transform: translateX(calc(-50% + 2px));
  -ms-transform: translateX(calc(-50% + 2px));
  transform: translateX(calc(-50% + 2px));
  pointer-events: none;
  box-sizing: content-box;
  border-width: 1px;
  border-style: solid;
  border-radius: 100px;
}

.emotion-2:before {
  top: -10px;
  left: -10px;
  right: -10px;
  bottom: -10px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: var(--pv-color-gray-9);
  -webkit-transition: opacity 200ms,background-color 200ms;
  transition: opacity 200ms,background-color 200ms;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <div
      aria-hidden="true"
      class="emotion-2"
    />
  </label>
</DocumentFragment>
`;

exports[`<Switch /> colors color "secondary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-7);
  height: var(--pv-size-base-4);
  border-radius: var(--pv-size-base-4);
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  overflow: hidden;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: var(--pv-color-gray-6);
}

.emotion-1:checked {
  background-color: var(--pv-color-secondary);
}

.emotion-1:checked+[aria-hidden] {
  -webkit-transform: translateX(calc(50% - 2px));
  -moz-transform: translateX(calc(50% - 2px));
  -ms-transform: translateX(calc(50% - 2px));
  transform: translateX(calc(50% - 2px));
}

.emotion-1:checked+[aria-hidden]:before {
  background-color: var(--pv-color-secondary-shade-2);
}

.emotion-1:disabled {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.4;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):hover+[aria-hidden]:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus+[aria-hidden]:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active+[aria-hidden]:before {
  opacity: 0.3;
}

.emotion-2 {
  background: var(--pv-color-secondary-contrast);
  border-color: var(--pv-color-gray-3);
  display: block;
  position: absolute;
  width: var(--pv-size-base-3);
  height: var(--pv-size-base-3);
  -webkit-transition: -webkit-transform 200ms;
  transition: transform 200ms;
  box-shadow: var(--pv-shadow-light-low);
  -webkit-transform: translateX(calc(-50% + 2px));
  -moz-transform: translateX(calc(-50% + 2px));
  -ms-transform: translateX(calc(-50% + 2px));
  transform: translateX(calc(-50% + 2px));
  pointer-events: none;
  box-sizing: content-box;
  border-width: 1px;
  border-style: solid;
  border-radius: 100px;
}

.emotion-2:before {
  top: -10px;
  left: -10px;
  right: -10px;
  bottom: -10px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: var(--pv-color-gray-9);
  -webkit-transition: opacity 200ms,background-color 200ms;
  transition: opacity 200ms,background-color 200ms;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <div
      aria-hidden="true"
      class="emotion-2"
    />
  </label>
</DocumentFragment>
`;

exports[`<Switch /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-7);
  height: var(--pv-size-base-4);
  border-radius: var(--pv-size-base-4);
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  overflow: hidden;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: var(--pv-color-gray-6);
}

.emotion-1:checked {
  background-color: var(--pv-color-primary);
}

.emotion-1:checked+[aria-hidden] {
  -webkit-transform: translateX(calc(50% - 2px));
  -moz-transform: translateX(calc(50% - 2px));
  -ms-transform: translateX(calc(50% - 2px));
  transform: translateX(calc(50% - 2px));
}

.emotion-1:checked+[aria-hidden]:before {
  background-color: var(--pv-color-primary-shade-2);
}

.emotion-1:disabled {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.4;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):hover+[aria-hidden]:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus+[aria-hidden]:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active+[aria-hidden]:before {
  opacity: 0.3;
}

.emotion-2 {
  background: var(--pv-color-primary-contrast);
  border-color: var(--pv-color-gray-3);
  display: block;
  position: absolute;
  width: var(--pv-size-base-3);
  height: var(--pv-size-base-3);
  -webkit-transition: -webkit-transform 200ms;
  transition: transform 200ms;
  box-shadow: var(--pv-shadow-light-low);
  -webkit-transform: translateX(calc(-50% + 2px));
  -moz-transform: translateX(calc(-50% + 2px));
  -ms-transform: translateX(calc(-50% + 2px));
  transform: translateX(calc(-50% + 2px));
  pointer-events: none;
  box-sizing: content-box;
  border-width: 1px;
  border-style: solid;
  border-radius: 100px;
}

.emotion-2:before {
  top: -10px;
  left: -10px;
  right: -10px;
  bottom: -10px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: var(--pv-color-gray-9);
  -webkit-transition: opacity 200ms,background-color 200ms;
  transition: opacity 200ms,background-color 200ms;
}

<label
    class="emotion-0 my-class-name"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <div
      aria-hidden="true"
      class="emotion-2"
    />
  </label>
</DocumentFragment>
`;

exports[`<Switch /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-7);
  height: var(--pv-size-base-4);
  border-radius: var(--pv-size-base-4);
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  overflow: hidden;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: var(--pv-color-gray-6);
}

.emotion-1:checked {
  background-color: var(--pv-color-primary);
}

.emotion-1:checked+[aria-hidden] {
  -webkit-transform: translateX(calc(50% - 2px));
  -moz-transform: translateX(calc(50% - 2px));
  -ms-transform: translateX(calc(50% - 2px));
  transform: translateX(calc(50% - 2px));
}

.emotion-1:checked+[aria-hidden]:before {
  background-color: var(--pv-color-primary-shade-2);
}

.emotion-1:disabled {
  cursor: not-allowed;
  pointer-events: none;
  opacity: 0.4;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):hover+[aria-hidden]:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus+[aria-hidden]:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active+[aria-hidden]:before {
  opacity: 0.3;
}

.emotion-2 {
  background: var(--pv-color-primary-contrast);
  border-color: var(--pv-color-gray-3);
  display: block;
  position: absolute;
  width: var(--pv-size-base-3);
  height: var(--pv-size-base-3);
  -webkit-transition: -webkit-transform 200ms;
  transition: transform 200ms;
  box-shadow: var(--pv-shadow-light-low);
  -webkit-transform: translateX(calc(-50% + 2px));
  -moz-transform: translateX(calc(-50% + 2px));
  -ms-transform: translateX(calc(-50% + 2px));
  transform: translateX(calc(-50% + 2px));
  pointer-events: none;
  box-sizing: content-box;
  border-width: 1px;
  border-style: solid;
  border-radius: 100px;
}

.emotion-2:before {
  top: -10px;
  left: -10px;
  right: -10px;
  bottom: -10px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: var(--pv-color-gray-9);
  -webkit-transition: opacity 200ms,background-color 200ms;
  transition: opacity 200ms,background-color 200ms;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <div
      aria-hidden="true"
      class="emotion-2"
    />
  </label>
</DocumentFragment>
`;
